utils/hints.go

```diff
@@ -239,8 +290,37 @@ func GenerateHints(annotations mapstr.M, container, prefix string, allSupportedH
 						if strings.HasPrefix(hintKey, container) {
 							// Split the key to get part[1] to be the hint
 							parts := strings.Split(hintKey, "/")
-							//We check whether the provided annotation follows the supported format and vocabulary. The check happens for annotations that have prefix co.elastic
-							found = checkSupportedHints(parts[1], allSupportedHints)
+
+							checkdigit := digitCheck.MatchString(parts[1]) // With this regex we check if enumeration for modules is provided
+							if checkdigit {
+								allSupportedHints = append(allSupportedHints, parts[1])
+
+								specificlist, _ := entries.GetValue(key)
+								if specificentries, ok := specificlist.(mapstr.M); ok {
+									for keyspec := range specificentries {
+										// enumeratedmodules will be populated only in cases we have module enumeration, like:
+										// "co.elastic.metrics/1.module":  "prometheus",
+										// "co.elastic.metrics/2.module":  "istiod",
+										enumeratedmodules = append(enumeratedmodules, keyspec)
+									}
+								}
+							}
+
+							if len(enumeratedmodules) > 0 {
```

Comment:
We don't need these conditions before the for cycle. Same for the ones below, it gets a bit confusing to read.

Reply:
Thank you! See [96897fd](https://github.com/elastic/elastic-agent-autodiscover/pull/83/commits/96897fd45e1e1169300dc37d27813992cf1df65b)

Reply:
Done